infrastructure/main.tf

```diff
@@ -60,6 +60,7 @@ locals {
     S2S_SECRET  = "${data.azurerm_key_vault_secret.s2s_secret.value}"
 
     ENVELOPES_QUEUE_CONNECTION_STRING           = "${data.terraform_remote_state.shared_infra.envelopes_queue_primary_listen_connection_string}"
+    ENVELOPES_QUEUE_MAX_DELIVERY_COUNT          = "${min(data.terraform_remote_state.shared_infra.envelopes_queue_max_delivery_count, var.envelopes_queue_max_delivery_count)}"
```

Comment:
I think it would be ideal to have this property based on the queue setting, i.e. make it the queue max delivery count minus some small constant, e.g. 3. We always want the setting in the application to be smaller than that in the queue (to make sure it's the app that dead-letters the message) and it will be easiest to manage it this way. That's what you had before, except it would be calculated in terraform instead of application code.